agent/src/agent.php

```diff
@@ -173,4 +182,41 @@
 
 $ingestDetails->hydrate();
 
+$signatureCheckTimer = $loop->addPeriodicTimer(60, static function () use ($loop, &$signatureCheckTimer, $ingest, $error, $info, $basePath, $expectedSignature) {
+    $signature = file_get_contents($basePath.'/signature.txt');
+
+    if ($signature === false) {
+        $error('Unable to read signature');
+
+        return;
+    }
```

Comment:
Thinking out loud here.

I wonder if this happens if we should consider this a change in signature.

What does this even mean.

I could imagine a world where we renamed the signature file. This should then trigger a restart.

If the file is removed, I feel like we would be better restarting and then letting the process fallover because the file is missing once it comes back online.